src/middlewared/middlewared/plugins/system.py

```diff
@@ -152,6 +156,335 @@ def debug(self, job):
+        )
+        ip6_address = data.get('guiv6address')
+        if (
+            ip6_address and
+            ip6_address != '::' and
+            ip6_address not in ip6_addresses_list
+        ):
+            verrors.add(
+                f'{schema}.guiv6address',
+                'Selected ipv6 address is not associated with this machine'
+            )
+
+        syslog_level = data.get('sysloglevel')
+        if not syslog_level:
+            verrors.add(
+                f'{schema}.sysloglevel',
+                'This field is required'
+            )
+
+        syslog_server = data.get('syslogserver')
+        if not syslog_server:
+            verrors.add(
+                f'{schema}.syslogserver',
+                'This field is required'
+            )
+        else:
+            match = re.match("^[\w\.\-]+(\:\d+)?$", syslog_server)
+            if not match:
+                verrors.add(
+                    f'{schema}.syslogserver',
+                    'Invalid syslog server format'
+                )
+            elif ':' in syslog_server:
+                port = int(syslog_server.split(':')[-1])
+                if port < 0 or port > 65535:
+                    verrors.add(
+                        f'{schema}.syslogserver',
+                        'Port specified should be between 0 - 65535'
+                    )
+
+        protocol = data.get('guiprotocol')
+        if not protocol:
+            verrors.add(
+                f'{schema}.guiprotocol',
+                'This field is required'
+            )
+        else:
+            if protocol != 'http':
+                certificate_id = data.get('guicertificate')
+                if not certificate_id:
+                    verrors.add(
+                        f'{schema}.guicertificate',
+                        'Protocol has been selected as HTTPS, certificate is required'
+                    )
+                else:
+                    certificate_obj = await self.middleware.call(
+                        'datastore.query',
+                        'system.certificate',
+                        [('id', '=', certificate_id)],
+                    )
+                    if len(certificate_obj) == 0:
+                        verrors.add(
+                            f'{schema}.guicertificate',
+                            'No matching certificate found in database records, kindly check again'
+                        )
+                        raise verrors
+
+                    certificate_obj = certificate_obj[0]
+                    # getting fingerprint for certificate
+                    fingerprint = await self.get_certificate_fingerprint(certificate_obj['cert_certificate'])
+                    if fingerprint:
+                        self.logger.debug('Fingerprint of the certificate used in the GUI: ' + fingerprint)
+        return verrors
+
+    @accepts(
+        Dict(
+            'general_settings',
+            IPAddr('guiaddress'),
```

Comment:
We dont have to match the attributes of legacy UI.

Itt requires some tweaking but I believe we could translate attribute names, gui* looks ugly.

What if we use `ui_*` instead, e.g.:
ui_certificate
ui_port
ui_protocol